packages/assets-controllers/src/crypto-compare-service/crypto-compare.ts

```diff
@@ -0,0 +1,154 @@
+ */
+function getMultiPricingURL(
+  fsyms: string,
+  tsyms: string,
+  includeUSDRate = false,
+) {
+  const updatedTsyms =
+    includeUSDRate && !tsyms.includes('USD') ? `${tsyms},USD` : tsyms;
+
+  const params = new URLSearchParams();
+  params.append('fsyms', fsyms);
+  params.append('tsyms', updatedTsyms);
+
+  const url = new URL(`${CRYPTO_COMPARE_DOMAIN}/data/pricemulti`);
+  url.search = params.toString();
+
+  return url.toString();
+}
+
+/**
+ * Handles an error response from the CryptoCompare API.
+ * Expected error response format
+ * { Response: "Error", Message: "...", HasWarning: false }
+ *
+ * @param json - The JSON response from the CryptoCompare API.
+ * @param json.Response - The response status.
+ * @param json.Message - The error message.
+ */
+function handleErrorResponse(json: { Response?: string; Message?: string }) {
+  if (json.Response === 'Error') {
+    throw new Error(json.Message);
+  }
+}
+
+/**
+ * Fetches the exchange rate for a given currency.
+ *
+ * @param currency - ISO 4217 currency code.
+ * @param nativeCurrency - Symbol for base asset.
+ * @param includeUSDRate - Whether to add the USD rate to the fetch.
+ * @returns Promise resolving to exchange rate for given currency.
+ */
+export async function fetchExchangeRate(
+  currency: string,
+  nativeCurrency: string,
+  includeUSDRate?: boolean,
+): Promise<{
+  conversionRate: number;
+  usdConversionRate: number;
+}> {
+  const json = await handleFetch(
+    getPricingURL(currency, nativeCurrency, includeUSDRate),
+  );
+
+  handleErrorResponse(json);
+  const conversionRate = Number(json[currency.toUpperCase()]);
+
+  const usdConversionRate = Number(json.USD);
+  if (!Number.isFinite(conversionRate)) {
+    throw new Error(
+      `Invalid response for ${currency.toUpperCase()}: ${
+        json[currency.toUpperCase()]
+      }`,
+    );
+  }
+
+  if (includeUSDRate && !Number.isFinite(usdConversionRate)) {
+    throw new Error(`Invalid response for usdConversionRate: ${json.USD}`);
+  }
+
+  return {
+    conversionRate,
+    usdConversionRate,
+  };
+}
+
+/**
+ * Fetches the exchange rates for multiple currencies.
+ *
+ * @param currency - The currency of the rates (ISO 4217).
+ * @param cryptocurrencies - The cryptocurrencies to get conversion rates for. Min length: 1. Max length: 300.
+ * @param includeUSDRate - Whether to add the USD rate to the fetch.
+ * @returns Promise resolving to exchange rates for given currencies.
+ */
+export async function fetchMultiExchangeRate(
+  currency: string,
+  cryptocurrencies: string[],
+  includeUSDRate: boolean,
+): Promise<Record<string, Record<string, string | null>>> {
+  const url = getMultiPricingURL(
+    Object.values(cryptocurrencies).join(','),
+    currency,
+    includeUSDRate,
+  );
+  const response = await handleFetch(url);
+  handleErrorResponse(response);
+
+  const rates: Record<string, Record<string, string | null>> = {};
+  for (const [key, value] of Object.entries(response) as [
```

Comment:
Maybe we could use `cryptocurrency` instead of `key`? an use `values` instead of `value` (so that we know it's a mapping for `cryptocurrency` to a fiat value, WDYT?

Reply:
Done in [8d40f6e](https://github.com/MetaMask/core/pull/4242/commits/8d40f6ed507e556869aa262075d282553f80abc8)